pkg/genlib/generator_interface.go

```diff
@@ -567,7 +567,48 @@ func fuzzyInt(previous int64, fuzziness, min, max float64) int64 {
 	return customRand.Int63n(int64(math.Ceil(higherBound-lowerBound))) + int64(lowerBound)
 }
 
+func fuzzyIntCounter(previous int64, fuzziness float64) int64 {
+	lowerBound := float64(previous)
+	higherBound := float64(previous) * (1 + fuzziness)
+	return customRand.Int63n(int64(math.Ceil(higherBound-lowerBound))) + int64(lowerBound)
+}
+
 func bindLong(fieldCfg ConfigField, field Field, fieldMap map[string]any) error {
+	if err := fieldCfg.ValidCounter(); err != nil {
+		return err
+	}
+
+	if fieldCfg.Counter {
+		var emitFNotReturn emitFNotReturn
+		emitFNotReturn = func(state *genState, buf *bytes.Buffer) error {
+			var previous int64
+			var dummyInt int64
+			var dummyFunc func() int64
+
+			if previousDummyInt, ok := state.prevCache[field.Name].(int64); ok {
+				previous = previousDummyInt
+			}
+
+			if fieldCfg.Fuzziness <= 0 {
+				dummyFunc = makeIntCounterFunc(previous, field)
+
+				dummyInt = dummyFunc()
+			} else {
+				dummyInt = fuzzyIntCounter(previous, fieldCfg.Fuzziness)
+			}
+
+			state.prevCache[field.Name] = dummyInt
+			v := make([]byte, 0, 32)
+			v = strconv.AppendInt(v, dummyInt, 10)
```

Comment:
I must confess, this code looks quite complicated with the buffer and strconv but it seems a pattern that is already used.

Reply:
The same for me :)
It was introduced with the huge performance refactoring at https://github.com/elastic/elastic-integration-corpus-generator-tool/pull/15